2.0/README.md

```diff
@@ -135,3 +143,34 @@ docker exec -it <CONTAINER_ID> /bin/bash
 
 After you [Docker exec](http://docker.io) into the running container, your current
 directory is set to `/opt/app-root/src`, where the source code is located.
+
+Performance tuning
+---------------------
+You can tune the number of threads per worker using the
+`PUMA_MIN_THREADS` and `PUMA_MAX_THREADS` environment variables.
+Additionally, the number of worker processes is determined by the number of CPU
+cores that the container has available, as recommended by
+[Puma](https://github.com/puma/puma)'s documentation. This is determined using
+the cgroup [cpusets](https://www.kernel.org/doc/Documentation/cgroup-v1/cpusets.txt)
+subsystem. You can specify the cores that the container is allowed to use by passing
+the `--cpuset-cpus` parameter to the [Docker](http://docker.io) run command:
+```
+$ docker run -e PUMA_MAX_THREADS=32 --cpuset-cpus='0-2,3,5' -p 8080:8080 sinatra-app
+```
+The number of workers is also limited by the memory limit that is enforced using
+cgroups. The builder image assumes that you will need 50 MiB as a base and
+another 15 MiB for every worker thread plus 0.1 MiB for each thread. Note that
```

Comment:
> every worker thread

The worker is a process, isn't it?

I'd expect this to be "x MiB as a base, another y MiB for every worker process, plus z MiB for each thread."